### ISSUE TITLE
Combining piped streams swallows data/events

### ISSUE THREAD

**Issue by vicatcu** [joepie91/node-combined-stream2#6]
I'm trying to combine streams from multiple json-parse-stream streams, and here is a fairly minimal example:

```javascript
var fs = require('fs');
var CombinedStream = require('combined-stream2');
var parseStream = require('json-parse-stream');

var combinedStream = CombinedStream.create();
var a = fs.createReadStream('a.json').pipe(parseStream());
var b = fs.createReadStream('b.json').pipe(parseStream());
combinedStream.append(a);
combinedStream.append(b);

// combinedStream.on('data', (data) => console.log('data-combined:', data)); 
// alternate tset, uncomment above line, comment below two lines
a.on('data', (data) => console.log('data-a', data));
b.on('data', (data) => console.log('data-b', data))
```

I only ever see `data-a` and `data-b` events,  and, switching around the comments, I get no `data-combined` events. If I remove the .pipe(parseStream()) chain from stream `a` like so:

```javascript
var fs = require('fs');
var CombinedStream = require('combined-stream2');
var parseStream = require('json-parse-stream');

var combinedStream = CombinedStream.create();
var a = fs.createReadStream('a.json');
var b = fs.createReadStream('b.json').pipe(parseStream());
combinedStream.append(a);
combinedStream.append(b);

combinedStream.on('data', (data) => console.log('data-combined:', data)); 
// alternate tset, uncomment above line, comment below two lines
// a.on('data', (data) => console.log('data-a', data));
// b.on('data', (data) => console.log('data-b', data))
```

I *do* get data-combined events, and if I remove .pipe(parseStream()) from both a and b, then I get more data-combined events. 

So basically, I think the symptom is when I try and combine piped streams the data doesn't get captured / emitted by the combined streams. I suspect this is not a bug in the library and that my expectations are flawed, but perhaps someone could shed light on how to do what I'm trying to do correctly?

**Comment by vicatcu** [comment]
I did find this on Stack Overflow http://stackoverflow.com/questions/17471659/creating-a-node-js-stream-from-two-piped-streams that seems pretty closely related. Maybe the solution there is something that this library could support as an option somehow.
